docs/translating.md

```diff
@@ -0,0 +1,169 @@
+# UI translation
+
+The website UI is automatically translated in [available languages](TODO: link
+to translation.ts). This is achieved with
+[`react-i18next`](https://react.i18next.com/).
+
+---
+
+<!-- vim-markdown-toc GitLab -->
+
+* [Workflows](#workflows)
+    * [Editing the source code](#editing-the-source-code)
+    * [Adding a new language](#adding-a-new-language)
+* [Architecture](#architecture)
+* [Available scripts](#available-scripts)
+    * [`generate-ui.js`](#generate-uijs)
+    * [`translate-ui.js`](#translate-uijs)
+    * [`translate-md.js`](#translate-mdjs)
+    * [`translate-faq.js`](#translate-faqjs)
+    * [`translate-releases.js`](#translate-releasesjs)
+
+<!-- vim-markdown-toc -->
+
+---
+
+## Workflows
+
+### Editing the source code
+
+When adding a new string containing a content which need to be translated,
+you need to wrap it inside a [`Trans`](https://react.i18next.com/latest/trans-component) React component
+or inside a function call to [`t`](https://react.i18next.com/latest/usetranslation-hook).
+
+Once the content modified,
+
+1. you can update the resource files with the [`generate-ui.js`](#generate-ui.js) script.
+    <details>
+    <summary>Script output example</summary>
+
+    ```
+    > npm run generate:ui
+
+    > nosgestesclimat-site@2.1.0 generate:ui
+    > node scripts/i18n/generate-ui.js
+
+    Static analysis of the source code...
+    Adding missing entries...
+    + Added 2 translations:
+        Cacher l'objectif
+        catégorie complétée
+    ~ Updated 1 translations:
+        components.stats.StatsContent.enSavoirPlus
+    - Missing 2 translations:
+        components.conversation.select.NumberedMosaic.choixAFaire
+        components.conversation.select.NumberedMosaic.choixEnTrop
+    Writting resources in /home/emile/Projects/datagir/nosgestesclimat-site/source/locales/ui/ui-fr.json...
+
+    ```
+    </details>
+
+    For _added_ and _updated_ entries, their translation are automatically
+    retrieved from the source code. However, for missing entries, you need
+    to replace the `'NO_TRANSLATION'` string by the adequate one.
+
+2. then, you need to translate the generated resource file into the other supported languages.
+This can be done automatically by using [`translate-ui.js`](#translate-ui.js).
+
+For non _hardcoded_ content such as the FAQ or Markdown pages.
+You need to modify the corresponding file inside `.source/locales/pages/` (resp. `.source/locales/faq`)
+and then running the script [`translate-md.js`](#translate-md.js) (resp. [`translate-faq.js`](#translate-faq.js))
+to translate it in the remaining languages.
```

Comment:
What about texts that where translated by deepl ? Are they tagged as "[auto]" or something, letting the user remove this [auto] tag when the write a better trnaslation ? Will the translation script check if this [auto] tag is there to avoid overwriting a better custom translation  ?

Reply:
For now, only missing entries or no up to date ones are translated. It means if a translation has been modified by hand and the corresponding source text is updated in the source code, the translation will be overridden. Then, the user will see a git diff and will decide to keep or not the translation.

Reply:
Let's see how it behaves in real subsequent updates.

Reply:
This thing about git diff could be in the doc :+1: